src/vm/diagnosticserver.cpp

```diff
@@ -0,0 +1,128 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "common.h"
+#include "diagnosticserver.h"
+#include "diagnosticsipc.h"
+#include "eventpipe.h"
+#include "eventpipeconfiguration.h"
+#include "processdescriptor.h"
+#include "sampleprofiler.h"
+
+#ifdef FEATURE_PAL
+#include "pal.h"
+#endif // FEATURE_PAL
+
+#ifdef FEATURE_PERFTRACING
+
+static DWORD WINAPI DiagnosticsServerThread(LPVOID /*lpThreadParameter*/)
+{
+    CONTRACTL
+    {
+        THROWS;
+        GC_TRIGGERS;
+        MODE_ANY;
+    }
+    CONTRACTL_END;
+
+    const ProcessDescriptor pd = ProcessDescriptor::FromCurrentProcess();
+    IpcStream::DiagnosticsIpc ipc("dotnetcore-diagnostic", pd.m_Pid);
+
+    while (true)
+    {
+        IpcStream *pStream = ipc.Accept();  // FIXME: Ideally this would be something like a std::shared_ptr
+        assert(pStream != nullptr);
```

Comment:
It seems like `DiagnosticsIpc::Accept` can return a nullptr on error but here we're doing an assert that it's not a nullptr

Reply:
This would happen when there is an error accepting a client, and in which case we move on (continue). Should we handle it differently?

Reply:
No, just that on Debug/Checked builds this `assert(pStream != nullptr)` will fire before we even continue. 😄

Reply:
I understand, should I remove it? I've been using it while debugging my clients failing :)

Reply:
I'll remove it.